pkg/sql/drop_type.go

```diff
@@ -249,6 +249,43 @@ func (p *planner) removeBackRefsFromAllTypesInTable(
 	return p.removeTypeBackReferences(ctx, typeIDs, desc.ID, jobDesc)
 }
 
+func (p *planner) addBackRefsFromAllTypesInType(ctx context.Context, desc *typedesc.Mutable) error {
+	typeIDs, err := desc.GetIDClosure()
+	if err != nil {
+		return err
+	}
+	for id := range typeIDs {
+		if id == desc.ID {
+			// Don't add a self back reference.
+			continue
+		}
+		jobDesc := fmt.Sprintf("updating type back reference %d for type %d", id, desc.ID)
+		if err := p.addTypeBackReference(ctx, id, desc.ID, jobDesc); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (p *planner) removeBackRefsFromAllTypesInType(
+	ctx context.Context, desc *typedesc.Mutable,
+) error {
+	typeIDMap, err := desc.GetIDClosure()
+	if err != nil {
+		return err
+	}
+	typeIDs := make([]descpb.ID, 0, len(typeIDMap))
+	for id := range typeIDMap {
+		if id == desc.ID {
+			// Don't add a self back reference.
+			continue
+		}
+		typeIDs = append(typeIDs, id)
+	}
```

Comment:
nit: consider using `catalog.DescriptorIDSet` here.

Reply:
Sorry, I overlooked the fact that `.GetIDClosure()` returned a map and not a slice (or even better, a `catalog.DescriptorIDSet`). My bad!